src/test/java/com/atlassian/db/replica/api/TestDualConnection.java

```diff
@@ -783,25 +783,39 @@ public void shouldStopReadOnlyMode() throws SQLException {
         connection.setReadOnly(false);
 
         assertThat(connectionProvider.getProvidedConnectionTypes())
-            .containsOnly(MAIN);
+            .containsOnly(REPLICA);
         verify(connectionProvider.singleProvidedConnection()).setReadOnly(false);
         assertThat(connection.isReadOnly()).isFalse();
     }
 
     @Test
-    public void shouldNotSwitchBackToReplica() throws SQLException {
+    public void shouldUtiliseReplicaEvenAfterDisablingReadOnly() throws SQLException {
         final ConnectionProviderMock connectionProvider = new ConnectionProviderMock();
         final Connection connection = DualConnection.builder(
             connectionProvider,
             permanentConsistency().build()
         ).build();
 
         connection.setReadOnly(false);
-        connection.setReadOnly(true);
+        connection.prepareStatement(SIMPLE_QUERY).executeQuery();
+
+        assertThat(connectionProvider.getProvidedConnectionTypes())
+            .containsOnly(REPLICA);
+    }
+
+    @Test
+    public void shouldNotSwitchBackToReplica() throws SQLException {
```

Comment:
This test case seems unrelated to the PR. Correct?

Reply:
Yes. The test case is unrelated. The diff doesn't show it, but   `shouldNotSwitchBackToReplica` was reusing `setReadOnly` to switch between the main database and replica. It will not work after the fix, so I changed it to use `executeUpdate` and `executeQuery`.